pkg/sql/opt/optbuilder/testdata/upsert

```diff
@@ -78,7 +78,7 @@ UPDATE SET a=5
 ----
 upsert abc
  ├── columns: <none>
- ├── canary column: 15
+ ├── canary column: 12
```

Comment:
The canary column is changing because of this change in `optbuilder/insert.go`:

```diff
- canaryScopeCol := &fetchScope.cols[findNotNullIndexCol(mb.tab.Index(cat.PrimaryIndex))]
+ canaryScopeCol := &fetchScope.cols[findNotNullIndexCol(index)]
```

The changes in these tests looks fine — a different not-null column is found. In this case, instead of the primary key, its the indexed non-null column `a`. I don't think this should not affect the results or performance of the `UPSERT` in any way.